### ISSUE TITLE
ubuntu-upstart: rsyslog and cron work fine -- but not with `start`

### ISSUE THREAD

**Issue by solidsnack** [docker-library/official-images#1080]
Starting a container and using `docker exec -it "$id" bash -il` to get a root shell, it's possible to start `rsyslog` and `cron` both: just run `rsyslog` and then `cron`.

But the Upstart commands fail:

```bash
root@aabbccddeeff:/# start rsyslog
start: Job failed to start
root@aabbccddeeff:/# start cron
start: Job failed to start
```

This seems strange and very solvable. I'm interested in this for the same reason many people other are: symmetry between small and very large environments.

**Comment by jpetazzo** [comment]
Which versions are you using?

I tried with image `acce1511` (buntu-upstart:latest as I write those lines) on Engine 1.8.2, using the AUFS storage driver; and it tells me that the services are already running:

```
jp@vorpal:~$ docker exec -ti 751 bash
root@7517927b58db:/# service ssh status
ssh start/running, process 41
root@7517927b58db:/# start rsyslog
start: Job is already running: rsyslog
root@7517927b58db:/# start cron
start: Job is already running: cron
```

And indeed they seem to be:

```
root@7517927b58db:/# ps faux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root        99  0.1  0.0  18176  3284 ?        Ss   18:22   0:00 bash
root       171  0.0  0.0  15568  2232 ?        R+   18:22   0:00  \_ ps faux
root         1  0.2  0.0  32728  3376 ?        Ss+  18:22   0:00 /sbin/init
root        38  0.0  0.0  23660  2280 ?        Ss   18:22   0:00 cron
root        46  0.0  0.0  61372  5336 ?        Ss   18:22   0:00 /usr/sbin/sshd -D
syslog      95  0.3  0.0 116580  2776 ?        Ssl  18:22   0:00 rsyslogd
root@7517927b58db:/#
```

**Comment by solidsnack** [comment]
My versions seem to be the exact same as yours:

```bash
:;  docker --version
Docker version 1.8.2, build 0a8c2e3
:;  docker images | fgrep ubuntu-upstart
ubuntu-upstart            latest              acce1511ff80        4 weeks ago         253.5 MB
```

The results of a run are not the same, though:

```bash
:;  docker run --name ubuntu ubuntu-upstart
##  In another terminal
:;  docker exec -it ubuntu ps faux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root        77  0.0  0.0  15560  1100 ?        Rs+  01:31   0:00 ps faux
root         1  0.0  0.0  32720  2000 ?        Ss   01:28   0:00 /sbin/init
root        40  0.0  0.0  61364  3060 ?        Ss   01:28   0:00 /usr/sbin/sshd
```

**Comment by solidsnack** [comment]
Not sure if host OS version makes any difference:
```bash
:;  lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.3 LTS
Release:        14.04
Codename:       trusty
```

**Comment by jpetazzo** [comment]
Super weird! Can you add the output of `docker info` and `docker version` ? (The latter will also display the Engine version, which is assume is 1.8.2, but given how strange this issue is, I can't rule out anything!)

**Comment by solidsnack** [comment]
```bash
:;  docker info
Containers: 4
Images: 28
Storage Driver: aufs
 Root Dir: /tmp/docker/aufs
 Backing Filesystem: extfs
 Dirs: 36
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: syslog
Kernel Version: 3.13.0-63-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 4
Total Memory: 7.305 GiB
Name: 0x01.docker.infra.fernet.io
ID: MO3H:ZSKF:VMKA:RM3G:2SV2:S7VA:VOC6:MV2D:FIVK:X6G2:Q2JN:KWPQ
WARNING: No swap limit support
:;  docker version
Client:
 Version:      1.8.2
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0a8c2e3
 Built:        Thu Sep 10 19:19:00 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.2
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0a8c2e3
 Built:        Thu Sep 10 19:19:00 UTC 2015
 OS/Arch:      linux/amd64
```

**Comment by jpetazzo** [comment]
For reference, mine is:

```
Containers: 6
Images: 972
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 988
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux jessie/sid
CPUs: 4
Total Memory: 11.46 GiB
Name: vorpal
ID: EAOW:L22K:PDB5:2J7H:TFXC:PG5H:2GFI:VUOV:EIBM:BMQN:T37G:UKGD
Username: jpetazzo
Registry: https://index.docker.io/v1/
```

The only differences I see are:
- kernel version
- dirperm1 support
- logging driver

None of those should be a problem, though.

Annoyingly, I can't find a way to tell upstart to be more verbose. 

If you `docker exec` into the container, is there anything in `/var/log` and `/var/log/upstart` ? (Check file modification timestamps, as some files might be already present in the image and not relevant to our interests here...)

I'll try to reproduce later on an Ubuntu 14.04 machine.

**Comment by dronnix** [comment]
Have the same problem at Docker 1.9.0

**Comment by chuyeow** [comment]
I had the same problem and after some tinkering around with the AppArmor profile that containers run with (in `/etc/apparmor.d/docker`), I realized that it's caused by ptrace being denied.

Adding this line to `/etc/apparmor.d/docker` and reloading the profile (with `apparmor_parser -r /etc/apparmor.d/docker` works:

```
ptrace (trace,read) peer=docker-default,
```

This is fixed in Docker master and in v1.10.2 but only at the point when Docker is installed (see https://github.com/docker/docker/pull/18393 and with AppArmor version >= 2.8.95). So for your existing Docker installation, I suggest adding the line manually.
